src/pyinit.jl

```diff
@@ -87,15 +99,7 @@ function __init__()
 
     if !already_inited
         # some modules (e.g. IPython) expect sys.argv to be set
-        if pyversion.major < 3
-            argv_s = ""
-            argv = unsafe_convert(Ptr{UInt8}, argv_s)
-            ccall(@pysym(:PySys_SetArgvEx), Void, (Cint,Ptr{Ptr{UInt8}},Cint), 1, &argv, 0)
-        else
-            argv_s = Cwchar_t[0]
-            argv   = unsafe_convert(Ptr{Cwchar_t}, argv_s)
-            ccall(@pysym(:PySys_SetArgvEx), Void, (Cint, Ptr{Ptr{Cwchar_t}}, Cint), 1, &argv, 0)
-        end
+        ccall(@pysym(:PySys_SetArgvEx), Void, (Cint, Ptr{UInt32}, Cint), 1, EmptyStringList(), 0)
```

Comment:
Isn't there a simpler way to do this?

Reply:
Keeping the branch and using `Ptr{Cstring}`, `Ptr{Cwstring}` and pass in an actual array of string should work too. I did it this way since,

1. This eliminate the branch.
2. So the code is actually not much longer
3. I want to see how easy it is to implement sth like this properly
4. It doesn't allocate anything (optimized out on 0.7) (yeah, I know it's just `__init__`)

When 0.6 support is dropped, this can be done with `@gc_preserve` without external definition.

```julia
ref0 = Ref{UInt32}(0)
@gc_preserve ref0 ccall(@pysym(:PySys_SetArgvEx), Void, (Cint, Ref{Ptr{Void}}, Cint), 1, pointer_from_objref(ref0), 0)
```

Reply:
The branch and the performance are irrelevant here.   But as your code shows, we don't actually need the branch, since a pointer to a pointer to an array of a 32-bit zero will work in both Python 2 and 3 (even though Python 2 needs only one byte of zero).

I would prefer to just use `Ptr{Cwstring}` to pass an an actual array rather than declaring a custom type here.